packages/dd-trace/src/opentracing/propagation/text_map.js

```diff
@@ -214,9 +216,43 @@ class TextMapPropagator {
     return this._config.tracePropagationStyle[mode].includes(name)
   }
 
+  _ensureTraceContextTakePrecedence (traceContext, spanContext, carrier) {
+    if (traceContext !== null && spanContext.toTraceId(true) === traceContext.toTraceId(true) &&
+        spanContext.toSpanId() !== traceContext.toSpanId()) {
+      if (ddParentIdTagKey in traceContext._trace.tags && traceContext._trace.tags[ddParentIdTagKey] !==
+          zeroHex) {
+        // tracecontext headers contain a p value, ensure this value is sent to backend
+        spanContext._trace.tags[ddParentIdTagKey] = traceContext._trace.tags[ddParentIdTagKey]
+      } else {
+        // if p value is not present in tracestate, use the parent id from the datadog headers
+        const ddCtx = this._extractDatadogContext(carrier)
+        if (ddCtx !== null) {
+          spanContext._trace.tags[ddParentIdTagKey] = ddCtx._spanId.toString().padStart(16, '0')
+        }
+      }
+      // the span_id in tracecontext takes precedence over the first extracted propagation style
+      spanContext._spanId = traceContext._spanId
+    }
+    // the first extracted context will be used to propagate the distributed trace
+    return spanContext
+  }
+
   _extractSpanContext (carrier) {
+    let spanContext = null
     for (const extractor of this._config.tracePropagationStyle.extract) {
-      let spanContext = null
+      // add logic to ensure tracecontext headers takes precedence over other extracted headers
+      if (spanContext !== null) {
+        if (this._config.tracePropagationExtractFirst) {
+          return spanContext
+        }
+        if (extractor !== 'tracecontext') {
+          continue
+        }
+        spanContext = this._ensureTraceContextTakePrecedence(
```

Comment:
nit: Tracecontext only takes precedence if we detect a conflicting spans in the same trace. We should try to make that clear (with a common or in the function name). 

```suggestion
        spanContext = this._resolveTraceContextConflicts(
```